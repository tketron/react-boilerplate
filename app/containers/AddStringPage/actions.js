/*
 *
 * AddStringPage actions
 *
 */

import { ADD_STRING, ADD_STRING_SUCCESS, ADD_STRING_ERROR } from './constants';

export function addNewString(s) {
  return {
    type: ADD_STRING,
    newString: s,
  };
}

export function addNewStringSuccess() {
  return {
    type: ADD_STRING_SUCCESS,
  };
}

export function addNewStringError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
