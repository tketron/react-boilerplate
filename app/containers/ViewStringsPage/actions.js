import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
} from './constants';

export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}

export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

export function stringsLoadingError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}
