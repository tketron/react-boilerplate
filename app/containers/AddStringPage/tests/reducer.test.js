import addStringPageReducer from '../reducer';
import { addNewString } from '../actions';

describe('addStringPageReducer', () => {
  it('returns the initial state', () => {
    expect(addStringPageReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the addString action', () => {
    expect(
      addStringPageReducer(undefined, addNewString('test string')),
    ).toMatchSnapshot();
  });
});
