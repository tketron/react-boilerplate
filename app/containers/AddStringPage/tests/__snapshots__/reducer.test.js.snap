// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addStringPageReducer handles the addString action 1`] = `
Immutable.Map {
  "newString": "test string",
}
`;

exports[`addStringPageReducer returns the initial state 1`] = `
Immutable.Map {
  "newString": "",
}
`;
